code/game/machinery/OpTable.dm

```diff
@@ -52,53 +52,50 @@
 /obj/machinery/optable/MouseDrop_T(atom/movable/O, mob/user)
 	return take_patient(O, user)
 
-/// Updates `patient` to be a carbon mob occupying the table, and returns it
-/obj/machinery/optable/proc/update_patient()
-	if(patient in buckled_mobs)
-		return patient // Current patient is still here, no need to look
-
-	patient = null
-
-	if(length(buckled_mobs))
-		for(var/mob/living/carbon/C in buckled_mobs)
-			patient = C
-
-			if(length(injected_reagents))
-				to_chat(C, "<span class='danger'>You feel a series of tiny pricks!</span>")
-
-			break
-
-	if(!no_icon_updates)
-		if(patient && patient.pulse)
-			icon_state = "table2-active"
-		else
-			icon_state = "table2-idle"
-
-	return patient
+/obj/machinery/optable/post_unbuckle_mob(mob/living/M)
+	. = ..()
+	if(M == patient)
+		patient = null
+		update_appearance(UPDATE_ICON_STATE)
 
 /obj/machinery/optable/proc/take_patient(mob/living/carbon/new_patient, mob/living/carbon/user)
 	if((!ishuman(user) && !isrobot(user)) || !istype(new_patient))
 		return
-	if(update_patient())
-		to_chat(usr, "<span class='notice'>The table is already occupied!</span>")
+	if(patient in buckled_mobs)
```

Comment:
Would we rather check if theres any buckled mobs at all? (I'm not sure)
```suggestion
	if(has_buckled_mobs())
```

Reply:
I mostly want to see if we're re-buckling the patient here, so I think this is intentional.